Better logging and notification of setting on fire

optimumtact opened the pull request:
This also logs bump fire to the gamelog incase someone ever needs to
find out who set who on fire with bumping (left out of admin
notification) as it would be very spammy

optimumtact committed:
Better logging and notification of setting on fire

This also logs bump fire to the gamelog incase someone ever needs to
find out who set who on fire with bumping (left out of admin
notification) as it would be very spammy

Jordie0608 commented on code/game/objects/items/weapons/cigs_lighters.dm:
```diff
@@ -73,6 +73,8 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	if(!isliving(M))
 		return
+	message_admins("[key_name_admin(user)] set (key_name_admin(M)) on fire")
+	log_game("[key_name(user)] set [key_name_user(M)] on fire")
```
`[key_name(M)]`

phil235 commented on code/game/objects/items/weapons/tools.dm:
```diff
@@ -319,6 +319,8 @@
 		if(isliving(O))
 			var/mob/living/L = O
+			message_admins("[key_name_admin(user)] set (key_name_admin(L)) on fire")
+			log_game("[key_name(user)] set [key_name(L)] on fire")
```
the mob isn't necessarily set on fire. It needs to have flammable fluid on them to be set on fire...

optimumtact replied:
Hmm that's a good point

KorPhaeron replied:
Check if they have firestacks

optimumtact committed:
Refactors IgniteMob to return true if the mob was set on fire

Also moves the DNA species ignite mob check to a better proc name to
make it clearer that it checks for a mob species that is valid to set on
fire

optimumtact committed:
Ensure we only log and notify admins when person is set on fire

This takes advantage of the now refactored IgniteMob returning true when
it successfully ignites the mob

optimumtact committed:
Explicitly return parent value

This prevents it returning false incorrectly

optimumtact commented:
I've refactored IgniteMob to return true when it has set the user on fire and taken advantage of this to only log and notify when the mob is actually truly set on fire

@tgstation/commit-access
